### PR TITLE
[entropy_src/dv] More efficient config coverage

### PR DESCRIPTION
**Pull request by martin-lueker**
This commit adds enhancments to the RNG vseq to perform more reconfigs per iteration and thus close more cover points with less simulation time.

- Identifies certain configurations as "Silent" meaning that they are blocked either by OTP or by inconsistent CSR configurations. These do not generate seeds and thus do not close any configuration CPs.
  - A new max_silent_reconfig_time parameter has been added to prompt a faster reconfig for these silent configurations.

- Identifies a bug in which random reconfigurations were not getting changed in Xcelium (it seems that if randomize() is called twice on the same object, the randomize routine is called, but the object
  is not updated in the calling task!).   This is fixed by creating a
  new copy of the dut_cfg object each time before randomizing.

- Removes long pauses when reconfiguring the DUT

- Removes the "Do NOT Disturb" flag in the RNG vseq. This debugging feature was incorrectly implemented and thus was preventing most reconfig trigger events it is also not needed anyway

- Includes a number of small constraint or knob changes to get more efficient coverage.

Signed-off-by: Martin Lueker-Boden <martin.lueker-boden@wdc.com>

### CODE REVIEW

File: hw/ip/entropy_src/dv/env/entropy_src_env_cfg.sv [hunk 1]
```diff
@@ -278,4 +281,20 @@ class entropy_src_env_cfg extends cip_base_env_cfg #(.RAL_T(entropy_src_reg_bloc
     `DV_CHECK(induce_targeted_transition_pct <= 100);
   endfunction
 
+  // Some combinations of environment and DUT configurations do not generate seeds, this function
+  // helps vseqs identify these inactive configurations to more quickly prompt a reconfiguation
+  // to get more coverage in a given run/
```
Comment (matutem) [inline hw/ip/entropy_src/dv/env/entropy_src_env_cfg.sv:286]:
```suggestion
  // helps vseqs identify these inactive configurations to more quickly prompt a reconfiguration
  // to get more coverage in a given run.
```

File: hw/ip/entropy_src/dv/env/entropy_src_env_cfg.sv [hunk 2]
```diff
@@ -278,4 +281,20 @@ class entropy_src_env_cfg extends cip_base_env_cfg #(.RAL_T(entropy_src_reg_bloc
     `DV_CHECK(induce_targeted_transition_pct <= 100);
   endfunction
 
+  // Some combinations of environment and DUT configurations do not generate seeds, this function
```
Comment (matutem) [inline hw/ip/entropy_src/dv/env/entropy_src_env_cfg.sv:284]:
```suggestion
  // Some combinations of environment and DUT configurations do not generate seeds. This function
```

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_base_vseq.sv [hunk 3]
```diff
@@ -24,7 +24,7 @@ class entropy_src_base_vseq extends cip_base_vseq #(
 
   virtual entropy_src_cov_if   cov_vif;
 
-  realtime default_cfg_pause = 50us;
+  realtime default_cfg_pause = 0;
```
Comment (matutem) [inline hw/ip/entropy_src/dv/env/seq_lib/entropy_src_base_vseq.sv:27]: Make it a localparam?

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv [hunk 4]
```diff
@@ -377,7 +386,7 @@ class entropy_src_rng_vseq extends entropy_src_base_vseq;
     if (cfg.mean_rand_csr_alert_time > 0) begin
       sched_csr_alert_time = randomize_failure_time(cfg.mean_rand_csr_alert_time);
       // randomize_failure_time uses negative values to communicate errors
-      `uvm_info(`gfn, $sformatf("sched_csr_alert_time: %g", sched_csr_alert_time), UVM_FULL)
+      `uvm_info(`gfn, $sformatf("sched_csr_alert_time: %g", sched_csr_alert_time/1ms), UVM_FULL)
```
Comment (matutem) [inline hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv:389]: Add units in the message string? Like "..._time: %g ms"?

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv [hunk 5]
```diff
@@ -698,18 +730,23 @@ class entropy_src_rng_vseq extends entropy_src_base_vseq;
...
       if (do_reset) begin
         apply_reset(.kind("HARD_DUT_ONLY"));
         post_apply_reset(.reset_kind("HARD"));
       end
       wait(cfg.under_reset == 0);
       do_reenable = 0;
 
+      `uvm_info(`gfn, "Generating new random config", UVM_MEDIUM)
+      // Make a new copy of the DUT cfg.
+      altcfg=new cfg.dut_cfg;
```
Comment (matutem) [inline hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv:749]:
```suggestion
      altcfg = new cfg.dut_cfg;
```

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv (hunk 4 above)
Comment (martin-lueker) [reply hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv:389]: Ah, good catch.

**Review by mwbranstad — COMMENTED** [review]
LGTM

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_base_vseq.sv (hunk 3 above)
Comment (martin-lueker) [reply hw/ip/entropy_src/dv/env/seq_lib/entropy_src_base_vseq.sv]: Created an environment configuration for this.

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv (hunk 4 above)
Comment (martin-lueker) [reply hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv]: Added units: ms .

**Comment by martin-lueker** [comment]
Hi @matutem, could you please take a look and see if I responded adequately to your comments?

**Comment by martin-lueker** [comment]
(Oh and thanks BTW for the review!)

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv [hunk 6]
```diff
@@ -678,15 +696,32 @@ class entropy_src_rng_vseq extends entropy_src_base_vseq;
...
+      if(!do_background_procs) begin
+        msg  = $sformatf("Reconfig pre-empted after %0.2f ms", ($realtime()-init_time)/1ms);
         `uvm_info(`gfn, msg, UVM_MEDIUM)
       end
-      // Stop other background procs (if they haven't been already)
-      // This will prompt a reconfig
-      do_background_procs = 0;
+       // Stop other background procs (if they haven't been already)
+       // This will prompt a reconfig
+       do_background_procs = 0;
```
Comment (matutem) [inline hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv:724]: Why the change in indentation?

**Review by matutem — APPROVED** [review]
I flagged a simple indentation issue, otherwise LGTM.

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv (hunk 6 above)
Comment (martin-lueker) [reply hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv:724]: Good question.  I will fix it.  Thank you!

**Comment by martin-lueker** [comment]
Thank you Guillermo! For the review!

File: hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv (hunk 6 above)
Comment (martin-lueker) [inline hw/ip/entropy_src/dv/env/seq_lib/entropy_src_rng_vseq.sv:724]:
```suggestion
      // Stop other background procs (if they haven't been already)
      // This will prompt a reconfig
      do_background_procs = 0;
```
